code/modules/mining/machine_vending.dm

```diff
@@ -159,47 +159,61 @@
 
 /obj/machinery/mineral/equipment_vendor/attackby(obj/item/I, mob/user, params)
 	if(istype(I, /obj/item/mining_voucher))
-		RedeemVoucher(I, user)
+		redeem_voucher(I, user)
 		return
 	if(default_deconstruction_screwdriver(user, "mining-open", "mining", I))
 		return
 	if(default_deconstruction_crowbar(I))
 		return
 	return ..()
 
-/obj/machinery/mineral/equipment_vendor/proc/RedeemVoucher(obj/item/mining_voucher/voucher, mob/redeemer)
-	var/items = list("Survival Capsule and Explorer's Webbing", "Resonator Kit", "Minebot Kit", "Extraction and Rescue Kit", "Crusher Kit", "Mining Conscription Kit")
+/**
+ * Allows user to redeem a mining voucher for one set of a mining equipment
+ *
+ * * Arguments:
+ * * voucher The mining voucher that is being used to redeem the mining equipment
+ * * redeemer The mob that is redeeming the mining equipment
+ */
+/obj/machinery/mineral/equipment_vendor/proc/redeem_voucher(obj/item/mining_voucher/voucher, mob/redeemer)
+	var/static/list/set_types = list()
+	var/list/items = list()
+	for(var/datum/voucher_set/current_set as anything in subtypesof(/datum/voucher_set))
+		set_types[initial(current_set.name)] = new current_set
```

Comment:
`set_types` is a static list but you seem to populate it with a full list of new `voucher_set` datums every time this proc is called.

It seems like `set_types` should only be setup once and the `items` list is the only thing that needs to be populated every redemption?

Reply:
You are right, I changed it so that the static list is populated only once now.